docs/proposals/scale_group_upgrade.md

```diff
@@ -0,0 +1,55 @@
+# Proposal Title
+
+## Description
+With the direction of moving to golden images (AMIs), there is a need to design the
+upgrade path for scale groups.
+
+## Rationale
+We currently provide an upgrade path for in-place upgrades.  We also need to provide a path
+for golden images when placed in scale groups.
+
+## Design
+The core principle here is to take advantage of the immutable infrastructure that is provided
+when working with golden images.
+
+The current playbooks in openshift ansible create autoscaling groups for infra/compute and 
+allow for custom groups as well.  This code lives in `role/openshift_aws`.
+
+Assume we are upgrading the compute node group:
+compute 123
+
+The upgrade process would look like this:
+* Create a new scale group `compute 456`
+* Scale up the new scale group to match the first groups capacity
+* Allow nodes to join the cluster by approving node CSRs
+* Set the nodes from `compute 123` as unschedulable
+* Drain the `compute 123` pods from the nodes
+* Remove scale group `compute 123`
```

Comment:
Would this be in the same upgrade playbook or in a separate playbook?

It seems that if it's in the same playbook then the proposed roll back strategy below isn't quite correct.

Reply:
@twiest,

The playbooks can be in the same or different.  I could see the rollback strategy either as mentioned, scaling the old ones back up or recreating the old scale groups with the old  launch config.  Either way would have the same overall effect.

I see your point in that we would need a go or no-go type of decision to occur before the last step of 'Remove scale group' occurred.  This might be difficult automating as I'm not sure who/what makes that determination.  I'll add a note.